rfcs/0021-tvmc_add_support_for_micro_targets.md

```diff
@@ -0,0 +1,461 @@
+
+```bash
+$ tvmc micro build
+usage: tvmc micro build [-h] [-f] PROJECT_DIR {zephyr,arduino,template} ...
+tvmc micro build: error: the following arguments are required: PROJECT_DIR, platform
+
+$ tvmc micro build -h
+usage: tvmc micro build [-h] [-f] PROJECT_DIR {zephyr,arduino,template} ...
+
+positional arguments:
+  PROJECT_DIR           Project dir to build.
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -f, --force           Force rebuild.
+
+platforms:
+  {zephyr,arduino,template}
+                        you must selected a platform from the list. You can pass '-h' for a selected platform to list its options.
+    zephyr              select Zephyr platform.
+    arduino             select Arduino platform.
+    template            select an adhoc template.
+
+$ tvmc micro build /tmp/x200 zephyr
+usage: tvmc micro build PROJECT_DIR zephyr [--list-options] -o OPTION=VALUE [OPTION=VALUE ...]
+tvmc micro build PROJECT_DIR zephyr: error: the following arguments are required: -o
+
+$ tvmc micro build /tmp/x200 zephyr --list-options
+usage: tvmc micro build PROJECT_DIR zephyr [--list-options] -o OPTION=VALUE [OPTION=VALUE ...]
+
+optional arguments:
+  --list-options        show all options/values for selected platforms/template.
+  -o OPTION=VALUE [OPTION=VALUE ...]
+                        verbose={true, false}
+                          run build with verbose output.
+
+                        zephyr_base=ZEPHYR_BASE
+                          path to the zephyr base directory.
+
+                        zephyr_board={mimxrt1050_evk, mps2_an521, nrf5340dk_nrf5340_cpuapp,
+                                      nucleo_f746zg, nucleo_l4r5zi, qemu_cortex_r5, qemu_riscv32,
+                                      qemu_riscv64, qemu_x86, stm32f746g_disco}
+                          name of the Zephyr board to build for. (required)
+
+$ tvmc micro build /tmp/x200 zephyr -o zephyr_board=stm32f746g_disco
+$
+```
+
+3. `flash` subcommand:
+
+```bash
+$ tvmc micro flash
+usage: tvmc micro flash [-h] PROJECT_DIR {zephyr,arduino,template} ...
+tvmc micro flash: error: the following arguments are required: PROJECT_DIR, platform
+
+$ tvmc micro flash -h
+usage: tvmc micro flash [-h] PROJECT_DIR {zephyr,arduino,template} ...
+
+positional arguments:
+  PROJECT_DIR           Project dir with a image built.
+
+optional arguments:
+  -h, --help            show this help message and exit
+
+platforms:
+  {zephyr,arduino,template}
+                        you must selected a platform from the list. You can pass '-h' for a selected platform to list its options.
+    zephyr              select Zephyr platform.
+    arduino             select Arduino platform.
+    template            select an adhoc template.
+
+$ tvmc micro flash /tmp/x200 zephyr
+usage: tvmc micro flash PROJECT_DIR zephyr [--list-options] -o OPTION=VALUE [OPTION=VALUE ...]
+tvmc micro flash PROJECT_DIR zephyr: error: the following arguments are required: -o
```

Comment:
just curious: why is `-o` a required arg? seems it should be always optional (i agree zephyr_board is required for the zephyr `microtvm_api_server.py`, but as an API-level concept, it shouldn't always be required to specify `-o`).

Reply:
@areusch that's right it should not be always required. It was as a TODO in the Draft PR. I think that if one or more required options exist in the project then `-o` must be required, otherwise that should be optional. (actually now `--project-options` - I'll update the text accordingly to the reviews in the code that changed the name of that option). It's now done in https://github.com/apache/tvm/pull/9229/commits/93bb565a6b05a7306c430f154e47bf842c80ee5a 

As a reference, if all options in zephyr are optional, after that change `tvmc` will show  `--project-option` between `[ ]`  and also no options in the help text as `(required)`:

```
$ tvmc micro create-project --force /tmp/x46 ~/scripts/sine.tar zephyr --help
usage: tvmc micro create-project project_dir MLF zephyr [--project-option [OPTION=VALUE ...]] [-h]

optional arguments:
  --project-option [OPTION=VALUE ...]
                        extra_files_tar=EXTRA_FILES_TAR
                          if given, during generate_project, uncompress the tarball at this path into the project dir.
                        
                        project_type={aot_demo, host_driven}
                          type of project to generate.
                        
                        west_cmd=WEST_CMD
                          path to the west tool. If given, supersedes both the zephyr_base option and ZEPHYR_BASE environment variable. Defaults to '/usr/bin/python3 -m west'.
                        
                        zephyr_board={mimxrt1050_evk, mps2_an521, nrf5340dk_nrf5340_cpuapp,
                                     nucleo_f746zg, nucleo_l4r5zi, qemu_cortex_r5, qemu_riscv32,
                                     qemu_riscv64, qemu_x86, stm32f746g_disco}
                          name of the Zephyr board to build for.
                        
                        config_main_stack_size=CONFIG_MAIN_STACK_SIZE
                          sets CONFIG_MAIN_STACK_SIZE for Zephyr board.
                        
  -h, --help, --list-options
                        show this help message with platform-specific options and exit.
```